### PR TITLE
Fix for issue 8747: date field change causes an uncaught exception

### LINKED ISSUE #8747: date field with non date data generates error
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.6--2022-04-25--5c9d898 Linux 5.17.4-200.fc35.x86_64 amd64  Java 17.0.2  JavaFX 18+12

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. generate new entry
2. enter non date data in date field, e.g. an `b`

### Appendix

...
<details>

<summary>Log File</summary>

```
java.lang.NullPointerException: Cannot invoke "java.time.temporal.TemporalAccessor.query(java.time.temporal.TemporalQuery)" because "<parameter1>" is null
	at org.jabref@5.6.60000/org.jabref.gui.util.component.TemporalAccessorPicker.getLocalDate(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.gui.util.component.TemporalAccessorPicker$InternalConverter.fromString(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.gui.util.component.TemporalAccessorPicker$InternalConverter.fromString(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.control.DatePicker.commitValue(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.control.DatePicker.lambda$new$2(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.beans.property.ReadOnlyBooleanPropertyBase.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Node$FocusedProperty.notifyListeners(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene$12.invalidated(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene$KeyHandler.setFocusOwner(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene$KeyHandler.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Node.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.scene.control.behavior.TableViewBehaviorBase.mousePressed(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.6.60000/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.6.60000/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```

</details>

### PR DESCRIPTION
**Pull request by huangderek02**
Fixes #8747

## Short Description

In library mode "biblatex", when user edits the entry's "date" field with an invalid input and clicks somewhere, an uncaught exception of null pointer occurs.

## Checklist
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

## UI changes (takes time to load gif images)
* Before
 * Type invalid inputs: nothing happen
 * Click somewhere: an exception occured;
 * Press Enter: input disappears
312775146_691460315939346_8658484514782598519_n

* After
 * Type invalid inputs: show warning icon
 * Click somewhere: nothing happens
 * Press Enter: input disappears
313129752_1170515957176323_7279031655860662192_n

312829199_3262902150588889_342121938428642385_n

### CODE REVIEW

**Comment by xianghao-wang** [comment]
Hello, it is my first time contributing to this project. I am confused about the failures of Windows and MacOs' deployment tests because the errors seem irrelevant to the codes.

**Comment by Siedlerchr** [comment]
That is a nice improvement. Regarding invalid inputs, as biblatex supports a couple of more dates, see also https://github.com/JabRef/jabref/pull/9310 ) can you try to integrate this with JabRef's date time parsing class?

**Comment by xianghao-wang** [comment]
> That is a nice improvement. Regarding invalid inputs, as biblatex supports a couple of more dates, see also #9310 ) can you try to integrate this with JabRef's date time parsing class?
> 
> And please take a look at the reviewdog output (Files changed tab)

Hello, could you help have a look at the new change by adding Date integration to the default string converter? I think the string converter provided by `DateEditorViewModel` already have worked with Date class [code](https://github.com/JabRef/jabref/blob/main/src/main/java/org/jabref/gui/fieldeditors/DateEditorViewModel.java#L41).

File: src/main/java/org/jabref/gui/fieldeditors/DateEditorViewModel.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public TemporalAccessor fromString(String string) {
                 if (StringUtil.isNotBlank(string)) {
                     try {
                         return dateFormatter.parse(string);
-                    } catch (DateTimeParseException exception) {
+                    } catch (Exception exception) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/fieldeditors/DateEditorViewModel.java:39]: Why did you change this to a generic exception? Exception should be always the most specfic one

**Comment by Siedlerchr** [comment]
I tested your PR and it looks good so far. Please have a look at the failing checkstyle tests and also the exception handling.
I cannot push to your repo so I can't do this myself
